libs/docker-client/gradle/integrationTest.gradle.kts

```diff
@@ -0,0 +1,54 @@
+/*
+    Copyright 2017-2021 Charles Korn.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+val sourceSets = the<SourceSetContainer>()
+val mainAndTestCommon = sourceSets["main"].output + sourceSets["testCommon"].output
+
+val integrationTest by sourceSets.creating {
+    compileClasspath += mainAndTestCommon
+    runtimeClasspath += mainAndTestCommon
+}
+
+val integrationTestImplementation by configurations.getting {
+    extendsFrom(configurations["testImplementation"])
+}
```

Comment:
```suggestion
configurations["integrationTestImplementation"].extendsFrom(configurations["testImplementation"])
```

Reply:
That would not work as `integrationTestImplementation` is being referred to in `dependencies`.

Reply:
Right, makes sense, I completely missed that.